mobile/android/android-application-manager.ts

```diff
@@ -44,10 +44,16 @@ export class AndroidApplicationManager extends ApplicationManagerBase {
 	}
 
 	public async startApplication(appIdentifier: string): Promise<void> {
-		await this.adb.executeShellCommand(["monkey",
-			"-p", appIdentifier,
-			"-c", "android.intent.category.LAUNCHER",
-			"1"]);
+		const pmDumpOutput = await this.adb.executeShellCommand(["pm", "dump", appIdentifier, "|", "grep", "-A", "1", "MAIN"]);
+		const fullActivityNameRegExp = /([A-Za-z]{1}[A-Za-z\d_]*\.)*[A-Za-z][A-Za-z\d_]*\/([a-z][a-z_0-9]*\.)*[A-Z_]($[A-Z_]|[\w_])*/;
+		const activityMatch = new RegExp(fullActivityNameRegExp, "m");
+		const possibleIdentifier = activityMatch.exec(pmDumpOutput)[0];
```

Comment:
Yes, it will be unexpected behavior. My point is that if we start the default main activity, after we have failed to get it from pm dump, we will not break the CLI for the users who use the default main activity.
When we use the monkey tool we know that it will start the main activity, no matter how it's named. With the new logic we search for the main activity name with `grep -A 1` and I think it is possible that on different Android versions the output can be different. Also if we hit other problems with the new approach we will not be able to start any application if we don't use the default main activity.